tests/test_cxl_image_classification_host.py

```diff
@@ -0,0 +1,156 @@
+    host_mem_size = 0x8000  # Needs to be big enough to test cache eviction
+
+    host_name = "foo"
+    root_port_switch_type = ROOT_PORT_SWITCH_TYPE.PASS_THROUGH
+    memory_controller = RootComplexMemoryControllerConfig(host_mem_size, "foo.bin")
+    root_ports = [RootPortClientConfig(0, "localhost", switch_port)]
+    memory_ranges = [MemoryRange(MEMORY_RANGE_TYPE.DRAM, 0x0, host_mem_size)]
+
+    config = CxlImageClassificationHostConfig(
+        host_name,
+        0,
+        root_port_switch_type,
+        memory_controller,
+        memory_ranges,
+        root_ports,
+        coh_type=COH_POLICY_TYPE.DotCache,
+    )
+
+    host_manager = CxlHostManager(host_port=host_port, util_port=util_port)
+    host = CxlComplexHost(config)
+
+    pci_bus_driver = PciBusDriver(host.get_root_complex())
+    cxl_bus_driver = CxlBusDriver(pci_bus_driver, host.get_root_complex())
+    cxl_mem_driver = CxlMemDriver(cxl_bus_driver, host.get_root_complex())
+
+    start_tasks = [
+        asyncio.create_task(host.run()),
+        asyncio.create_task(host_manager.run()),
+        asyncio.create_task(sw_conn_manager.run()),
+        asyncio.create_task(physical_port_manager.run()),
+        asyncio.create_task(virtual_switch_manager.run()),
+        asyncio.create_task(dev1.run()),
+        asyncio.create_task(dev2.run()),
+        asyncio.create_task(dev3.run()),
+        asyncio.create_task(dev4.run()),
+    ]
+
+    wait_tasks = [
+        asyncio.create_task(sw_conn_manager.wait_for_ready()),
+        asyncio.create_task(physical_port_manager.wait_for_ready()),
+        asyncio.create_task(virtual_switch_manager.wait_for_ready()),
+        asyncio.create_task(host_manager.wait_for_ready()),
+        asyncio.create_task(host.wait_for_ready()),
+        asyncio.create_task(dev1.wait_for_ready()),
+        asyncio.create_task(dev2.wait_for_ready()),
+        asyncio.create_task(dev3.wait_for_ready()),
+        asyncio.create_task(dev4.wait_for_ready()),
+    ]
+    await asyncio.gather(*wait_tasks)
+
+    async def test_configs():
+        await pci_bus_driver.init()
+        await cxl_bus_driver.init()
+        await cxl_mem_driver.init()
+
+        logger.debug(cxl_mem_driver.get_devices())
+        for device in cxl_mem_driver.get_devices():
+            # NOTE: The list should match the dev order
+            # Not tested, though
+            # otherwise the dev base may not match the IRQ ports
+            host.append_dev_mmio_range(
+                device.pci_device_info.bars[0].base_address, device.pci_device_info.bars[0].size
+            )
+
+        # TODO: Not working right now. Make it work in the future.
+        # await host._host_simple_processor.store(hpa_base, 0x40, 0xAAAAAAAA)
+        # await dev1._cxl_type2_device.cxl_cache_readline(0x00000000)
+        # await dev1._cxl_type2_device._cxl_cache_manager.send_d2h_req_test()
+
+    test_tasks = [asyncio.create_task(test_configs()), asyncio.create_task(asyncio.sleep(4))]
```

Comment:
Please utilize `asyncio.run()` instead.

Reply:
Tried this, but I don't think using asyncio.run will work: 
RuntimeError: asyncio.run() cannot be called from a running event loop